playground/src/components/playground/PlaygroundInput.tsx

```diff
@@ -22,36 +32,66 @@ export function PlaygroundInput({
-    updateMethod: handleMethodChange,
-  } = useStudioStore("method", "path", "updatePath", "updateMethod");
+  const { activeRoute, setActiveRoute, appRoutes } = useStudioStore(
+    "activeRoute",
+    "setActiveRoute",
+    "appRoutes",
+  );
 
+  const path = useUrlPreview() ?? "";
   return (
     <form
       ref={formRef}
       onSubmit={onSubmit}
       // NOTE - For some reason, in prod (not locally), we get a `margin-block-end: 1rem` on the form element from the user-agent-stylesheet
       //        That's why we add `m-0` to the form element.
-      className="flex items-center justify-between bg-input border rounded-lg m-0"
+      className="flex items-center justify-between bg-input border rounded-lg mx-0 py-1"
     >
-      <div className="flex flex-grow items-center space-x-0">
-        <RequestMethodCombobox
-          method={method}
-          handleMethodChange={handleMethodChange}
-          allowUserToChange
-        />
-        <Input
-          type="text"
-          value={path}
-          onChange={(e) => {
-            handlePathInputChange(e.target.value);
-          }}
-          className="flex-grow text-xs w-full bg-transparent font-mono border-none shadow-none focus:ring-0 ml-0"
-        />
-      </div>
+      <FpDropdownMenu>
+        <FpDropdownMenuTrigger className="flex flex-grow items-center ml-2 gap-2 h-min hover:bg-muted data-[state=open]:bg-muted rounded-sm">
+          <Method
+            method={activeRoute?.method || "GET"}
+            className="ml-2 text-sm"
+          />
+          <div className="flex items-center w-full my-2 text-xs font-mono border-none shadow-none focus:ring-0 ml-0 py-0">
+            {path}
+          </div>
+          <CaretSortIcon className="w-3 h-3 mr-1" />
+        </FpDropdownMenuTrigger>
+        <FpDropdownMenuPortal>
+          <FpDropdownMenuContent align="start">
+            <FpDropdownMenuRadioGroup
+              value={activeRoute ? getRouteId(activeRoute) : ""}
+            >
+              {appRoutes.map((route) => (
+                <FpMinimalDropdownMenuRadioItem
+                  key={getRouteId(route)}
+                  onSelect={() => setActiveRoute(route)}
+                  value={getRouteId(route)}
+                  className={cn(
+                    "py-0.25",
+                    "focus:bg-muted",
+                    "aria-checked:bg-muted aria-checked:focus:text-accent-foreground",
+                  )}
+                >
+                  <div
+                    className={cn(
+                      "flex-grow grid grid-cols-[2.5rem_auto] items-center gap-1 px-1 rounded-md",
+                    )}
+                  >
+                    <Method method={route.method} className="ml-1 text-xs" />
+                    <div className="flex w-full my-2">
+                      <div className="flex-initial text-xs w-full bg-transparent font-mono border-none shadow-none focus:ring-0 ml-0 disabled:cursor-text disabled:bg-muted py-0">
```

Comment:
this will be tricky but we probably want to truncate this somehow, especially as the number of query params gets large

<img width="450" alt="image" src="https://github.com/user-attachments/assets/1ee0719f-6e89-4017-9ff7-af558c0f2706" />

Reply:
okay this is what i went with: 
<img width="460" alt="image" src="https://github.com/user-attachments/assets/289cb37f-2ae2-4c94-92ce-7bcae5d89da3" />

Reply:
maybe we do a ...... tooltip?!